src/command.rs

```diff
@@ -62,80 +65,100 @@ pub async fn init(kind: PackageType, name: Option<PackageName>) -> eyre::Result<
         dependencies: vec![],
     };
 
-    manifest.write().await?;
+    manifest
+        .write()
+        .await
+        .wrap_err("failed to write manifest file")?;
 
-    PackageStore::create().await
+    PackageStore::create()
+        .await
+        .wrap_err("failed to create buffrs project directories")
 }
 
 /// Adds a dependency to this project
-pub async fn add(registry: RegistryUri, dependency: &str) -> eyre::Result<()> {
+pub async fn add(registry: RegistryUri, dependency: &str) -> miette::Result<()> {
     let lower_kebab = |c: char| (c.is_lowercase() && c.is_ascii_alphabetic()) || c == '-';
 
     let (repository, dependency) = dependency
         .trim()
         .split_once('/')
-        .wrap_err("Invalid dependency specification")?;
+        .ok_or_else(|| miette!("locator {dependency} is missing a repository delimiter"))?;
 
     ensure!(
         repository.chars().all(lower_kebab),
-        "Repositories must be in lower kebab case"
+        "repository {repository} is not in kebab case"
     );
 
     let repository = repository.into();
 
     let (package, version) = dependency
         .split_once('@')
-        .wrap_err_with(|| format!("Invalid dependency specification: {dependency}"))?;
+        .ok_or_else(|| miette!("dependency specification is missing version part: {dependency}"))?;
 
     let package = package
         .parse::<PackageName>()
-        .wrap_err_with(|| format!("Invalid package name supplied: {package}"))?;
+        .wrap_err_with(|| format!("invalid package name: {package}"))?;
 
     let version = version
         .parse::<VersionReq>()
-        .wrap_err_with(|| format!("Invalid version requirement supplied: {package}"))?;
+        .into_diagnostic()
+        .wrap_err_with(|| format!("not a valid version requirement: {version}"))?;
 
-    let mut manifest = Manifest::read().await?;
+    let mut manifest = Manifest::read()
+        .await
+        .wrap_err("failed to read manifest file")?;
 
     manifest
         .dependencies
         .push(Dependency::new(registry, repository, package, version));
 
-    manifest.write().await
+    manifest
+        .write()
+        .await
+        .wrap_err("failed to write manifest file")
```

Comment:
```suggestion
        .wrap_err_with(|| miette!("failed to write manifest file"))
```

(would be even better to add the helpful details here)

Reply:
> (would be even better to add the helpful details here)
trying to keep the PR size under control but agree